src/frontend/packages/lib_video/src/components/common/VideoWidgetProvider/widgets/DeleteVideo/index.tsx

```diff
@@ -0,0 +1,124 @@
+  ConfirmationModal,
+  FoldableItem,
+  report,
+  useCurrentResourceContext,
+} from '@lib-video/../../lib_components/src';
+import { Button } from 'grommet';
+import { useState } from 'react';
+import toast from 'react-hot-toast';
+import { defineMessages, useIntl } from 'react-intl';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { useDeleteVideo } from '@lib-video/api/useDeleteVideo';
+import { useCurrentVideo } from '@lib-video/hooks';
+
+const messages = defineMessages({
+  info: {
+    defaultMessage:
+      'DANGER ZONE : This widget allows you to delete the VOD permanently.',
+    description: 'Info of the widget used for deleting the video.',
+    id: 'components.DeleteVideo.info',
+  },
+  title: {
+    defaultMessage: 'DANGER ZONE',
+    description: 'Title of the widget used for deleting the video.',
+    id: 'components.DeleteVideo.title',
+  },
+  confirmDeleteTitle: {
+    defaultMessage: 'Confirm delete video',
+    description: 'Title of the widget used for confirmation.',
+    id: 'components.DeleteVideoConfirm.title',
+  },
+  confirmDeleteText: {
+    defaultMessage:
+      'Are you sure you want to delete this VOD ? This action is irreversible.',
+    description: 'Text of the widget used for confirmation.',
+    id: 'components.DeleteVideoConfirm.text',
+  },
+  deleteButtonText: {
+    defaultMessage: 'Delete video',
+    description: 'Text of the delete button.',
+    id: 'components.DeleteVideoButton.text',
+  },
+  videoDeleteSuccess: {
+    defaultMessage: 'Video successfully deleted',
+    description: 'Text of the delete confirmation toast.',
+    id: 'components.deleteVideoSuccess.text',
+  },
+  videoDeleteError: {
+    defaultMessage: 'Failed to delete the video',
+    description: 'ext of the delete error toast..',
+    id: 'components.videoDeleteError.text',
+  },
+});
+
+const StyledAnchorButton = styled(Button)`
+  height: 50px;
+  font-family: 'Roboto-Medium';
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+export const DeleteVideo = () => {
+  const intl = useIntl();
+  const [context] = useCurrentResourceContext();
+  const history = useHistory();
+  const [showDeleteConfirmationModal, setShowDeleteConfirmationModal] =
+    useState(false);
+  const video = useCurrentVideo();
+  const deleteVideo = useDeleteVideo({
+    onSuccess: () => {
+      toast.success(intl.formatMessage(messages.videoDeleteSuccess), {
+        position: 'bottom-center',
+      });
+      setShowDeleteConfirmationModal(false);
+      history.push(context.resourcePath || `/my-contents/videos`);
+    },
+    onError: (err: unknown) => {
+      report(err);
+      toast.error(intl.formatMessage(messages.videoDeleteError), {
+        position: 'bottom-center',
+      });
+    },
+  });
+
+  if (!context.isFromWebsite) {
+    return null;
+  }
+
+  return (
+    <FoldableItem
+      infoText={intl.formatMessage(messages.info)}
+      initialOpenValue
```

Comment:
I don't understand where it is initialized and why it's working... There is some magic I don't see.

Reply:
Are you talking about the context state ?

Reply:
https://github.com/openfun/marsha/pull/2210/commits/4b28ef09757f8a21f54c7e5714a86f25ea08cd4e#diff-d5e7c8a2047ffd1d224e1b7a64dbc145099104138bd16469da576950b5fd1ee1